### PR TITLE
fix: Fix TS errors in `components` directory

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

Typescript errors are resolved in the `component` directory except for transaction based components

Video of running `yarn tsc`. Notice that no more TS errors are originating from `component` directory.

https://github.com/MetaMask/metamask-mobile/assets/10508597/12c1e77c-b25a-4d10-8626-260246550fb5

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/ConnectQRHardware/SelectQRAccounts/index.tsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ const SelectQRAccounts = (props: ISelectQRAccountsProps) => {
           <View style={[styles.account]}>
             <CheckBox
               style={[styles.checkBox]}
-              disabled={item.exist}
+              // disabled={item.exist}
```
Comment (sethkfman) [inline app/components/Views/ConnectQRHardware/SelectQRAccounts/index.tsx:141]: Delete if not needed

**Review by tommasini — DISMISSED** [review]
LGTM!

File: app/components/Views/ConnectQRHardware/SelectQRAccounts/index.tsx (hunk 1 above)
Comment (Cal-L) [reply app/components/Views/ConnectQRHardware/SelectQRAccounts/index.tsx]: Nice catch. This didn't need to be commented out or removed. I re-added it.

**Review by tommasini — APPROVED** [review]
LGTM

**Pull request merged**